### ISSUE TITLE
Better support for non-letters in option names.

### ISSUE THREAD

**Issue by schernichkin** [scallop/scallop#150]
Consider following example:
```
class Conf(arguments: Seq[String]) extends ScallopConf(arguments) {
  val `source-folder` = opt[String]
}
```
Since scallop uses identifier names to generate option names, it will infer `--output$minusfolder` for this option. But `--source-folder` is pretty legal in command-line. Why not demangle some legal characters? (replacement map can be found here: https://github.com/scala/scala/blob/2.12.x/src/library/scala/reflect/NameTransformer.scala)

**Comment by Rogach** [comment]
Yes, that makes sense. I'll implement the transformation.

However, I feel that using backtick-escaped values is a little cumbersome. Currently Scallop automatically transforms camelCase in vals into snake-case option names - so for example, the following will result in the same option (`--source-folder`):
```scala
val sourceFolder = opt[String]()
```

**Comment by Rogach** [comment]
Released `3.1.1` with this change, should sync to Maven Central within an hour.

**Comment by schernichkin** [comment]
Thanks. Your comment on camel case was useful though, I switched to camel case.
